File: core/src/ru/myitschool/dcrawler/skills/action/SimpleAttackAction.java
package ru.myitschool.dcrawler.skills.action;

import ru.myitschool.dcrawler.math.MathAction;
import ru.myitschool.dcrawler.math.NumberAction;
import ru.myitschool.dcrawler.skills.Skill;

public class SimpleAttackAction extends ActionAdapter {

    private MathAction action;
    private MathAction critAction;

    public SimpleAttackAction(Skill skill, MathAction action) {
        super(skill);
        this.action = action;
        this.critAction = new NumberAction(action.max());
    }

    @Override
    protected MathAction successDamage() {
        return action;
    }

    @Override
    protected MathAction critSuccessDamage(int standardDamage) {
        return critAction;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/action/SimpleAttackAction.kt
package com.wolg_vlad.dcrawler.entities.skills.action

import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.math.MathAction
import com.wolg_vlad.dcrawler.math.NumberAction

class SimpleAttackAction(skill: Skill, private val action: MathAction) : ActionAdapter(skill) {
    private val critAction: MathAction
    override fun successDamage(): MathAction? {
        return action
    }

    override fun critSuccessDamage(standardDamage: Int): MathAction? {
        return critAction
    }

    init {
        critAction = NumberAction(action.max())
    }
}